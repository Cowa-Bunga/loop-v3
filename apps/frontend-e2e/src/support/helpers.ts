export const testIdSelector = (testId: string) => `[data-testid="${testId}"]`

export const COOKIE_NAME = 'cowabunga-user-cookie'
