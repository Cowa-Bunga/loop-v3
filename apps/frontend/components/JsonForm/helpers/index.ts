const makeInputId = (id: string, suffix?: string) => {
  const parts = id.split('/')

  if (suffix) {
    return `${parts[parts.length - 1]}-${suffix}`
  }

  return parts[parts.length - 1]
}

export { makeInputId }
