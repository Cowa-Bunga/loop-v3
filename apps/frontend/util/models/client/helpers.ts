import { defaultsDeep } from 'lodash'
import { MODEL_CLIENT_EMPTY } from './model'

export const modelClientList = (
  data: { [s: string]: unknown } | ArrayLike<unknown>
): IClient[] => {
  return Object.entries(data)
    .map(modelClientData)
    .map((el: never) => el['1'])
    .sort((a: IClient, b: IClient) => a.name.localeCompare(b.name))
}

export const modelClientData = (data: unknown): IClient => {
  return defaultsDeep(data, MODEL_CLIENT_EMPTY)
}
