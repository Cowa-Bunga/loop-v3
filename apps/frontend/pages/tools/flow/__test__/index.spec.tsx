import IntelTools from '../index.page'
import { renderWithProviders } from '@test/helpers'

describe('IntelTools', function () {
  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(<IntelTools />)

    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
