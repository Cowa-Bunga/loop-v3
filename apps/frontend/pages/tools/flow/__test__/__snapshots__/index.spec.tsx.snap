// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntelTools should render successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-o0v5hs-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1qvppxn-MuiTypography-root"
          >
            <img
              alt="logo"
              src="https://www.loop.co.za/wp-content/uploads/2021/12/Logo.svg"
              style="margin-top: 5px; height: 20px;"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LockIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root css-1abb3vo"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          style="height: calc(100vh - 60px); width: 100vw; padding-top: 20px;"
        >
          <div
            class="react-flow"
            data-testid="rf__wrapper"
            style="width: 100%; height: 100%; overflow: hidden; position: relative; z-index: 0;"
          >
            <div
              class="react-flow__renderer"
              style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
            >
              <div
                class="react-flow__pane"
                style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
              >
                <div
                  class="react-flow__viewport react-flow__container"
                  style="transform: translate(0px,0px) scale(1);"
                >
                  <svg
                    class="react-flow__edges react-flow__container"
                    height="500"
                    style="z-index: 0;"
                    width="500"
                  >
                    <defs />
                    <g />
                  </svg>
                  <div
                    class="react-flow__edgelabel-renderer"
                  />
                  <div
                    class="react-flow__nodes"
                    style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
                  >
                    <div
                      aria-describedby="react-flow__node-desc-1"
                      class="react-flow__node react-flow__node-customTitle nopan selectable"
                      data-id="1"
                      data-testid="rf__node-1"
                      role="button"
                      style="z-index: 0; transform: translate(400px,100px); pointer-events: all; visibility: hidden;"
                      tabindex="0"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-w3opwq-MuiPaper-root-MuiCard-root"
                      >
                        <div
                          class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                          data-handlepos="top"
                          data-id="1-null-target"
                          data-nodeid="1"
                        />
                        <h4
                          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
                        >
                          Company
                        </h4>
                        <div
                          class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                          data-handlepos="bottom"
                          data-id="1-null-source"
                          data-nodeid="1"
                        />
                      </div>
                    </div>
                    <div
                      aria-describedby="react-flow__node-desc-1"
                      class="react-flow__node react-flow__node-customImg nopan selectable"
                      data-id="2"
                      data-testid="rf__node-2"
                      role="button"
                      style="z-index: 0; transform: translate(500px,300px); pointer-events: all; visibility: hidden;"
                      tabindex="0"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1d6vb52-MuiPaper-root-MuiCard-root"
                      >
                        <div
                          class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                          data-handlepos="top"
                          data-id="2-null-target"
                          data-nodeid="2"
                        />
                        Region
                        <div
                          class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                          data-handlepos="bottom"
                          data-id="2-null-source"
                          data-nodeid="2"
                        />
                      </div>
                    </div>
                    <div
                      aria-describedby="react-flow__node-desc-1"
                      class="react-flow__node react-flow__node-custom nopan selectable"
                      data-id="3"
                      data-testid="rf__node-3"
                      role="button"
                      style="z-index: 0; transform: translate(500px,500px); pointer-events: all; visibility: hidden;"
                      tabindex="0"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1kix3wt-MuiPaper-root-MuiCard-root"
                      >
                        <div
                          class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                          data-handlepos="top"
                          data-id="3-null-target"
                          data-nodeid="3"
                        />
                        Hub
                        <div
                          class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                          data-handlepos="bottom"
                          data-id="3-null-source"
                          data-nodeid="3"
                        />
                      </div>
                    </div>
                    <div
                      aria-describedby="react-flow__node-desc-1"
                      class="react-flow__node react-flow__node-custom nopan selectable"
                      data-id="4"
                      data-testid="rf__node-4"
                      role="button"
                      style="z-index: 0; transform: translate(600px,700px); pointer-events: all; visibility: hidden;"
                      tabindex="0"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1kix3wt-MuiPaper-root-MuiCard-root"
                      >
                        <div
                          class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                          data-handlepos="top"
                          data-id="4-null-target"
                          data-nodeid="4"
                        />
                        Driver
                        <div
                          class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                          data-handlepos="bottom"
                          data-id="4-null-source"
                          data-nodeid="4"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <svg
              class="react-flow__background"
              data-testid="rf__background"
              style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
            >
              <pattern
                height="20"
                id="pattern-1undefined"
                patternTransform="translate(-0.5,-0.5)"
                patternUnits="userSpaceOnUse"
                width="20"
                x="0"
                y="0"
              >
                <circle
                  cx="0.5"
                  cy="0.5"
                  fill="#91919a"
                  r="0.5"
                />
              </pattern>
              <rect
                fill="url(#pattern-1undefined)"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
            </svg>
            <div
              class="react-flow__panel react-flow__controls bottom left"
              data-testid="rf__controls"
              style="pointer-events: all;"
            >
              <button
                aria-label="zoom in"
                class="react-flow__controls-button react-flow__controls-zoomin"
                title="zoom in"
                type="button"
              >
                <svg
                  viewBox="0 0 32 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
                  />
                </svg>
              </button>
              <button
                aria-label="zoom out"
                class="react-flow__controls-button react-flow__controls-zoomout"
                title="zoom out"
                type="button"
              >
                <svg
                  viewBox="0 0 32 5"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h32v4.2H0z"
                  />
                </svg>
              </button>
              <button
                aria-label="fit view"
                class="react-flow__controls-button react-flow__controls-fitview"
                title="fit view"
                type="button"
              >
                <svg
                  viewBox="0 0 32 30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94c-.531 0-.939-.4-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
                  />
                </svg>
              </button>
              <button
                aria-label="toggle interactivity"
                class="react-flow__controls-button react-flow__controls-interactive"
                title="toggle interactivity"
                type="button"
              >
                <svg
                  viewBox="0 0 25 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="react-flow__panel react-flow__minimap bottom right"
              data-testid="rf__minimap"
              style="pointer-events: all;"
            >
              <svg
                aria-labelledby="react-flow__minimap-desc-1"
                height="150"
                role="img"
                viewBox="-190.00000000000003 -23.333333333333336 980 746.6666666666666"
                width="200"
              >
                <title
                  id="react-flow__minimap-desc-1"
                >
                  React Flow mini map
                </title>
                <path
                  class="react-flow__minimap-mask"
                  d="M-213.33333333333337,-46.66666666666667h1026.6666666666667v793.3333333333333h-1026.6666666666667z
        M0,0h500v500h-500z"
                  fill="rgb(240, 240, 240, 0.6)"
                  fill-rule="evenodd"
                  pointer-events="none"
                  stroke="none"
                  stroke-width="1"
                />
              </svg>
            </div>
            <div
              class="react-flow__panel react-flow__attribution bottom right"
              data-message="Please only hide this attribution when you are subscribed to React Flow Pro: https://pro.reactflow.dev"
              style="pointer-events: all;"
            >
              <a
                aria-label="React Flow attribution"
                href="https://reactflow.dev"
                rel="noopener noreferrer"
                target="_blank"
              >
                React Flow
              </a>
            </div>
            <div
              id="react-flow__node-desc-1"
              style="display: none;"
            >
              Press enter or space to select a node.
              You can then use the arrow keys to move the node around.
               Press delete to remove it and escape to cancel.
               
            </div>
            <div
              id="react-flow__edge-desc-1"
              style="display: none;"
            >
              Press enter or space to select an edge. You can then press delete to remove it or escape to cancel.
            </div>
            <div
              aria-atomic="true"
              aria-live="assertive"
              id="react-flow__aria-live-1"
              style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(100%);"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
